pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP    *net.IPNet
+	Ports map[uint16]struct{}
+}
+
+// ParseBlacklist takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseBlacklist(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		conn := &ConnectionFilter{Ports: map[uint16]struct{}{}}
+
+		// retrieve valid IPs
+		if strings.Contains(ip, "/") {
+			_, subnet, err := net.ParseCIDR(ip)
+			if err != nil {
+				log.Debugf("Could not parse %s", err)
+			}
+			conn.IP = subnet
+		} else if strings.Contains(ip, ":") {
```

Comment:
https://golang.org/pkg/strings/#ContainsRune

Is a better fit here if you're only looking for one character